changelog/6757.feature.rst

```diff
@@ -0,0 +1 @@
+Added new assertion diff view when similar strings are compared.
```

Comment:
I think it's worth expanding a bit more.

```suggestion
When displaying a comparison between two texts in an assertion failure, if the texts differ significantly (more than 30% of lines and more than 4 lines), they are now displayed one after the other rather than in an interleaved diff (+/-) view.
In such cases, a diff is often incomprehensible, and it is easier to see the difference in full.
```

Reply:
Also I believe this file should be an improvement: `6757.improvement.rst`.

Reply:
Done.